### ISSUE TITLE
[4.0] Cannot add read more in editor of default install

### ISSUE THREAD

**Issue by PhilETaylor** [joomla/joomla-cms#20477]
### Steps to reproduce the issue

Edit article, select CMS Content dropdown, select read more.

### Expected result

read more added to WYSIWYG

### Actual result

JS Error

```
Uncaught TypeError: (intermediate value)(intermediate value)(intermediate value).match is not a function
```

![screen recording 2018-05-19 at 09 04 pm](https://user-images.githubusercontent.com/400092/40272627-4150e2d4-5ba8-11e8-8c41-72c4c1a64612.gif)

### System information (as much as possible)

79fd94942e1758cff32da7f8f368b5f78cc75c40
Google chrome

**Comment by brianteeman** [comment]
Could not replicate

![readmore](https://user-images.githubusercontent.com/1296369/40273561-f9c32dc0-5bb9-11e8-87b6-6aa8068e647e.gif)

**Comment by PhilETaylor** [comment]
replication from scratch:  
![screen recording 2018-05-19 at 11 14 pm](https://user-images.githubusercontent.com/400092/40273580-8071a77a-5bba-11e8-8460-0da97c37cdca.gif)
http://screenshot.myjoom.la/3Q0z2p0H3r1Z

**Comment by carlitorweb** [comment]
I can not replicate it either
![readmore](https://user-images.githubusercontent.com/3956472/40273692-7b08d76c-5b93-11e8-8465-97e6a6257d00.gif)

**Comment by PhilETaylor** [comment]
I can also replicate it in firefox

**Comment by infograf768** [comment]
I CAN replicate on Firefox and Safari. 
Those that can't should test from last staging.

Exactly the same error pointed by @PhilETaylor 
```
TypeError: (intermediate value).match is not a function[Learn More]
admin-article-readmore.min.js:1:161
window.insertReadmore
http://localhost:8888/newfolder/joomla40/media/com_content/js/admin-article-readmore.min.js:1:161
anonymous
http://localhost:8888/newfolder/joomla40/media/editors/tinymce/js/tinymce.min.js:3:1
setupEditor/</n.onclick
http://localhost:8888/newfolder/joomla40/media/editors/tinymce/js/tinymce.min.js:1:871
vm/r.fire
http://localhost:8888/newfolder/joomla40/media/vendor/tinymce/tinymce.min.js:1:187603
fire
http://localhost:8888/newfolder/joomla40/media/vendor/tinymce/themes/modern/theme.min.js:1:30800
a
http://localhost:8888/newfolder/joomla40/media/vendor/tinymce/themes/modern/theme.min.js:1:35026
u
http://localhost:8888/newfolder/joomla40/media/vendor/tinymce/tinymce.min.js:1:11719
C
http://localhost:8888/newfolder/joomla40/media/vendor/tinymce/tinymce.min.js:1:11870
l
http://localhost:8888/newfolder/joomla40/media/vendor/webcomponentsjs/js/webcomponents-hi-sd-ce.min.js:1:25807
```

**Comment by infograf768** [comment]
Setting to confirmed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20477">issues.joomla.org/tracker/joomla-cms/20477</a>.</sub>

**Comment by dgrammatiko** [comment]
Here is the code for admin-article-readmore.es6.js:
```js
(() => {
  'use strict';
  const options = window.Joomla.getOptions('xtd-readmore');

  window.insertReadmore = (editor) => {
    if (!options) {
      // Something went wrong!
      throw new Error(`XTD Button 'read more' not properly initialized`)
    }

    const content = window.Joomla.editors.instances[editor].getValue();

    if (content && !content.match(/<hr\s+id=("|')system-readmore("|')\s*\/*>/i)) {
      Joomla.editors.instances[editor].replaceSelection('<hr id="system-readmore">');
    } else {
      // TODO replace with joomla-alert
      alert(options.exists);
      return false;
    }
    return true;
  };
})();
```
If someone can make a pr I would really appreciate it 🙏

**Comment by carlitorweb** [comment]
Thank @infograf768, now yes. I can replicate it. I thought I had the last staging.

**Comment by infograf768** [comment]
@dgrammatiko 
Looks like we have to modify 
`content.match` to `editor.match`

Files concerned
/media/com_content/js/admin-article-readmore.es6.js
/media/com_content/js/admin-article-readmore.js

And then correct the minified files.

If I do that, no more errors.
Want me to do the PR?

NOTE: What we have in core is different from what you pasted above

**Comment by dgrammatiko** [comment]
> Looks like we have to modify content.match to editor.match

Nope, I've already provided the code (it's also tested)

**Comment by infograf768** [comment]
@dgrammatiko 
I just wanted to help. The issue is NOT here in the e6 but in the admin-article-readmore.js and minified
Please do as you wish.

**Comment by dgrammatiko** [comment]
the admin-article-readmore.js is generated from the es6 file, that's the source

**Comment by infograf768** [comment]
OK, then do it as I don't know how to.

**Comment by dgrammatiko** [comment]
BTW, with your proposal I guess you don't have an error but also you can insert as many read more (`hr`) as you like

**Comment by ghazal** [comment]
@dgrammatiko 
Looks like nobody made a PR with your code, 
that works alright BTW.

I can do it but I need some details:
if I understand properly, the modified code should be applied to:
admin-article-readmore.es6.js
??
what I don't get is how to minify it. Can I use a minifying tool online ?
Or else ?

**Comment by laoneo** [comment]
Yes in the es6 file. How to transpile to es5 and then minify can be found in #20396. You need npm installed.

**Comment by ghazal** [comment]
I postponed until now my digging into npm. Guess this is the opportunity.
Tks for your tuto.

**Comment by laoneo** [comment]
Thank your for stepping up to help.

**Comment by brianteeman** [comment]
closed see #21207
